src/mme-app/actionHandlers/defaultMmeProcedureActionHandlers.cpp

```diff
@@ -635,3 +628,58 @@ ActStatus ActionHandlers::default_s1_ho_handler(ControlBlock& cb)
     return ActStatus::PROCEED;
 }
 
+/******************************************************
+* Action handler : default_erab_mod_indication_handler
+*******************************************************/
+ActStatus ActionHandlers::default_erab_mod_indication_handler(ControlBlock& cb)
+{
+    log_msg(LOG_DEBUG, "default_erab_mod_indication_handler: Entry \n");
+
+    ProcedureStats::num_of_erab_mod_ind_received++;
+
+    UEContext *ueCtxt = static_cast<UEContext*>(cb.getPermDataBlock());
+    if (ueCtxt == NULL)
+    {
+        log_msg(LOG_ERROR, "UE Context is NULL \n");
+
+        MmeContextManagerUtils::deleteUEContext(cb.getCBIndex());
+        return ActStatus::HALT;
+    }
+
+    MsgBuffer *msgBuf = static_cast<MsgBuffer*>(cb.getMsgData());
+    if (msgBuf == NULL)
+    {
+        log_msg(LOG_DEBUG, "process_erab_mod_indication: msgBuf is NULL \n");
+        return ActStatus::PROCEED;
+    }
+
+    const s1_incoming_msg_data_t *msgData_p =
+            static_cast<const s1_incoming_msg_data_t*>(msgBuf->getDataPointer());
+    if (msgData_p == NULL)
+    {
+        log_msg(LOG_ERROR, "Failed to retrieve data buffer \n");
+        return ActStatus::PROCEED;
+    }
+
+    const struct erab_mod_ind_Q_msg &erabModInd =
+            (msgData_p->msg_data.erab_mod_ind_Q_msg_m);
+
+    MmeErabModIndProcedureCtxt *erabModIndProc_p =
+            MmeContextManagerUtils::allocateErabModIndProcedureCtxt(cb);
+    if (erabModIndProc_p != NULL)
+    {
+        erabModIndProc_p->setErabToBeModifiedList(
+                erabModInd.erab_to_be_mod_list.erab_to_be_mod_item,
+                erabModInd.erab_to_be_mod_list.count);
+
+        SM::Event evt(eRAB_MOD_IND_START, NULL);
+        cb.addEventToProcQ(evt);
+    }
+    else
+    {
+        log_msg(LOG_INFO, "Failed to allocate procedure context \n");
+    }
+
+    return ActStatus::PROCEED;
```

Comment:
In case of failure also we are returning PROCEED. How this should be handled ?

Reply:
Here failure maybe due to some memory congestion. We would want to ignore the current event/message and let eNB retry. 
The control goes back to state machine fwk module which picks the next event in the queue.